Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp

```diff
@@ -350,53 +350,42 @@ void JitArm64::fcmpx(UGeckoInstruction inst)
 	ARM64Reg VA = fpr.R(a, REG_IS_LOADED);
 	ARM64Reg VB = fpr.R(b, REG_IS_LOADED);
 
-	ARM64Reg WA = gpr.GetReg();
-	ARM64Reg XA = EncodeRegTo64(WA);
-
-	FixupBranch pNaN1, pNaN2, pNaN3, pLesser, pGreater;
+	FixupBranch pLesser, pGreater;
 	FixupBranch continue1, continue2, continue3;
-	ORR(XA, ZR, 32, 0, true);
 
 	if (a != b)
 	{
 		m_float_emit.FCMP(EncodeRegToDouble(VA), EncodeRegToDouble(VA));
 
 		// if (B != B) or (A != A), goto NaN's jump target
-		pNaN1 = B(CC_NEQ);
+		FALLBACK_IF(CC_NEQ);
```

Comment:
Yeah, Lioncash told me it wasn't a function but a macro - Didn't think to look it up before as it seemed very much like a conditional function, like if(); and B(); ..

Just in theory, are you sure FALLBACK_IF(); has to be done before m_float_emit.FCMP();? - Since it is currently working (all fcmpx calls just end up at the interpreter, but the interpreter handles them fine, though), even if they have been emitted by the JitArm64 first.

So now I'm trying to rework fcmpx to handle NaNs properly and to not having to fallback at all if possible.. So once again this PR can be ignored for the time being...